src/widgets/splits/SplitInput.cpp

```diff
@@ -569,7 +569,9 @@ void SplitInput::insertCompletionText(const QString &input_)
         else if (text[i] == '@')
         {
             input = "@" + input_ +
-                    (getSettings()->mentionUsersWithComma ? ", " : " ");
+                    (edit.isFirstWord() && getSettings()->mentionUsersWithComma
```

Comment:
With how `isFirstWord` is implemented, is this not equivalent to checking if `i == 0` here? Maybe we want the comma to be appended in more situations though.

Reply:
> With how isFirstWord is implemented, is this not equivalent to checking if i == 0 here?

No, in cases like: ` @zn<Tab>` `i == 1`.

> Maybe we want the comma to be appended in more situations though.

Could you expand? I can't think of in what situations we'd want to have comma appended.

Reply:
As discussed in chat: ![](https://cdn.zneix.eu/RkwxhJB.png)

I agree with this, maybe in the future we could upgrade / expand `isFirstWord()` into something that'd check if all words before the cursor are just mentions and append a comma.